src/System.Transactions/tests/NonMsdtcPromoterTests.cs

```diff
@@ -2337,23 +1989,14 @@ public static void TestCase_SetDistributedIdAtWrongTime()
                     try
                     {
                         SetDistributedTransactionId(dummyPSPE, Transaction.Current, guidToSet);
-                        TestFailed(testCaseDescription, failureDetail);
+                        bool shouldNotBeExecuted = true;
+                        Assert.False(shouldNotBeExecuted);
                     }
-                    catch (TransactionException ex)
+                    catch (Exception ex)
                     {
+                        Assert.True(ex is TransactionException || (ex is TargetInvocationException && ex.InnerException is TransactionException));
```

Comment:
ditto

---
In reply to: [82718954](https://github.com/dotnet/corefx/pull/12533#discussion_r82718954) [](ancestors = 82718954)